R/batchmark.R

```diff
@@ -0,0 +1,182 @@
+#' @title Run machine learning benchmarks as distributed experiments.
+#' 
+#' @description
+#' This function is a very parallel version of \code{\link{benchmark}}.
+#' Experiments are created in the provided registry for each combination of
+#' learners, tasks and resamplings. The experiments are then stored in a registry and the 
+#' runs can be started via \code{\link[batchtools]{submitJobs}]}. A job is one train/test split
+#' of the outer resampling. In case of nested resampling (e.g. with \code{\link{makeTuneWrapper}}), 
+#' each job is a full run of inner resampling, which can be parallelized in a second step 
+#' with \pkg{ParallelMap}. For details on the usage and support backends have 
+#' a look at the batchtools tutorial page: 
+#' \url{https://github.com/mllg/batchtools}.
+#' 
+#' The general workflow with \code{batchmark} looks like that:
+#' \itemize{
+#' \item{1.}{\code{batchtools::createExperimentRegistry()}}
+#' \item{2.}{\code{batchmark(...)}}
+#' \item{3.}{\code{batchtools:submitJobs()}}
+#' \item{3.5}{wait until the jobs are finished...}
+#' \item{4.}{\code{reduceBatchtoolsResult()}}
+#' }
+#' @inheritParams benchmark
+#' @param data.ids [\code{integer}]\cr
+#'   Dataset IDs to download from Open-ML. Default is none.
```

Comment:
I'm not sure that this is the right place. Is there a good argument against just getting the tasks before calling batchmark? 
```{r}
f = function(id) convertOMLTaskToMlr(getOMLTask(id))
tasks = lapply(data.ids, f)
```